content/en/docs/architecture/ci-operator-internals/observer-pods.md

```diff
@@ -0,0 +1,217 @@
+  # Only one of the following two should be specified
+  from: os # This image will be used to run the observer
+  # Literal image stream tag that will be used to run the observer
+  from_image:
+    ...
+  # This file has to be inside the same observer folder
+  commands: observer-commands.sh
+  resources:
+    requests:
+      cpu: 10m
+      memory: 10Mi
+  documentation: |-
+    A simple observer
+{{< / highlight >}}
+
+As it is for regular tests, observer's payload file must have `-commands` suffix (file extension has not
+been taken into account this time): 
+{{< highlight bash >}}
+#!/bin/bash
+
+set -o errexit
+set -o nounset
+set -o pipefail
+set -o xtrace
+
+echo "I'm going to silently observe your test"
+{{< / highlight >}}
+
+The final folder content wuold be, in this case:
+- `observer-observer.yaml`
+- `observer-commands.sh`
+
+## Execution Model
+Observers are started before any test from the `pre:` chain does and they keep staying alive till
+the last step from the `test:` chain completes.  
+
+When an observer fails, its failure will be reported inside logs and jUnit files but the test
+will keep going without being affected at all.  
+
+When the last test from `test:` chain finishes, a `SIGTERM` is sent to every observer and its related pod is deleted. After the test completes, no 'observer pods' pods will be found inside `ci-op-xxxx` namespace.
+
+Artifacts are being collected as they were regular tests.
+
+It's a good habit writing observer payload following this approach:
+{{< highlight bash >}}
+#!/bin/bash
+
+set -o errexit
+set -o nounset
+set -o pipefail
+set -o xtrace
+
+function cleanup() {
+    echo 'this will be executed last'
+}
+
+trap cleanup EXIT
+
+# Do something here
+# An observer could be killed in the middle of its execution!
+...
+{{< / highlight >}}
+## Features
+### Shared directory
+`$SHARED_DIR` is still being monted on observer pods but it is *read-only*.
+As regular tests modify it, kubernetes does the heavy lift of propagating the changes.
+
+The following example shows what to expect from the observer point of view when a test write something on
+the `$SHARED_DIR`.  
+
+Timeline of events:
+- Observer `observer-1` starts, `$SHARED_DIR` is empty
+- Test `test-1` starts
+- Test `test-1` writes file `dummy.txt` inside `$SHARED_DIR`
+- Test `test-1` ends gracefully
+- After a variable amount of time (up to ~2 minutes) kubernetes updates `$SHARED_DIR` content on `observer-1`
+- At this point only observer pod `observer-1` sees `dummy.txt`
+
+### Kubeconfig
+Observers will get a `read-write` kubeconfig as soon as it is provided by some tests or any other agent
+at the path specified in `$KUBECONFIG`.
+
+It is safe to check whether it exist or not doing so:
+{{< highlight bash >}}
+if [ -f "$KUBECONFIG" ]; then
+    ...
+fi
+{{< / highlight >}}
+
+Kubeconfig is treated differently from `$SHARED_DIR`, so observers could receive at any time. Please do not make any assumption about order of events.
```

Comment:
`kubeconfig` is treated differently from `$SHARED_DIR`, so observers could receive **it** at any time. Please do not make any **assumptions** about **the** order of events.